tests/unittest/_torch/modules/test_fused_activation_quant.py

```
@@ -0,0 +1,242 @@
+
+    sign_bit = (weight < 0).to(torch.uint8)
+    weight_abs = weight.abs()
+
+    ord_val = torch.searchsorted(E2M1_BOUNDS.to(device), weight_abs, out_int32=True).to(torch.uint8)
+    round_val = torch.any((weight_abs.unsqueeze(-1) == E2M1_BOUNDS.to(device)) * mask, dim=-1)
+    fp4_val = (sign_bit * 0b1000 + ord_val + round_val).to(torch.uint8)
+    return fp4_val
+
+
+def dequantize_fp4(
+    packed: torch.Tensor, scale: torch.Tensor, scale2: torch.Tensor, m: int, n: int
```

Ruff (0.14.14)

[warning] Unused function argument: `scale`

(ARG001)